cylc/flow/config.py

```diff
@@ -1896,12 +1904,82 @@ def get_graph_raw(
         graph_raw_edges.sort(key=lambda x: [y if y else '' for y in x[:2]])
         return graph_raw_edges
 
+    @staticmethod
+    def _detect_unsatisfiable_deps(gr_edges, is_validate, warn=True):
+        """Detect un-satisfiable tasks in a graph.
+
+        If not `is_validate` the gr_edge will be modified to mark it as
+        un-satisfiable. This transformation is used by cylc graph to change the
+        nodes appearance.
+        """
+        un_satisfiable_edges = []
+        for edges in gr_edges.values():
+            for ind, edge in enumerate(edges):
+                if not edge:
+                    continue
+                if is_validate:
+                    # gr_edges are in the is_validate format
+                    (
+                        (left_task, left_point),
+                        (right_task, right_point)
+                    ) = edge
+                    if not any(
+                        # is the task present in the graph at this cycle?
+                        (left_task, left_point) in {left, right}
+                        for left, right in gr_edges.get(left_point, [])
+                    ):
+                        # ... no, add it to the list
+                        un_satisfiable_edges.append((
+                            Tokens(
+                                cycle=str(left_point),
+                                task=left_task,
+                            ).relative_id,
+                            Tokens(
+                                cycle=str(right_point),
+                                task=right_task,
+                            ).relative_id,
+                        ))
+
+                else:
+                    # gr_edges are in the regular format
+                    left_id, right_id, _, *tail = edge
+                    ltokens = Tokens(left_id, relative=True)
+                    left_task = get_point(ltokens['task'])
+                    left_point = get_point(ltokens['cycle'])
+                    if not any(
+                        # is the task present in the graph at this cycle?
+                        left_id in {_left_id, _right_id}
+                        for _left_id, _right_id, *_ in gr_edges.get(
+                            left_point, []
+                        )
+                    ):
+                        # ... no, add it to the list
+                        un_satisfiable_edges.append((left_id, right_id))
+                        # mark the dependency as un-satisfiable
+                        # (used by cylc graph to change the nodes appearance)
+                        edges[ind] = (
+                            left_id,
+                            right_id,
+                            False,
+                            *tail
+                        )
+
+        if warn and un_satisfiable_edges:
+            LOG.warning(
```

Comment:
Is there any way of avoiding expanding this warning for every recurrence - IMO users would find
```bash
WARNING - Detected un-satisfiable dependencies:
    * foo[-PT1H] => foo [foo occurs at PT3H intervals]
```

more useful than

```bash
WARNING - Detected un-satisfiable dependencies:
    * 19831213T0800Z/foo => 19831213T0900Z/foo
    * 19831213T1100Z/foo => 19831213T1200Z/foo
```

Reply:
> Is there any way of avoiding expanding this warning for every recurrence

The exact cycle points might be useful debugging information to help the user determine which recurrence is causing the problem. E.G. if the problem cycle points are PT6H apart then the user knows to look for a recurrence with a PT6H pattern.